pkg/server/skydb/pq/builder/factory.go

```diff
@@ -26,6 +26,7 @@ import (
 type PredicateSqlizerFactory interface {
 	UpdateTypemap(typemap skydb.RecordSchema) skydb.RecordSchema
 	AddJoinsToSelectBuilder(q sq.SelectBuilder) sq.SelectBuilder
+	NewSort(s skydb.Sort) (string, error)
```

Comment:
This is a `predicate` `sqlizer` factory, and this new function fits neither.

Reply:
Think the original factory owns joinedTables and extraFields and they are also referenced by assembling sort expression for sorting by reference.
So decide to extends the original factory capability and rename the `PredicateSqlizerFactory` to `SqlizerFactory`